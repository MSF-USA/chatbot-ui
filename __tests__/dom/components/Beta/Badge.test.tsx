import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import '@testing-library/jest-dom';

import BetaBadge from '@/components/Beta/Badge';

describe('BetaBadge', () => {
  it('renders beta text', () => {
    render(<BetaBadge />);

    const badgeText = screen.getByText('Beta');
    expect(badgeText).toBeInTheDocument();
  });

  it('renders as a span element', () => {
    const { container } = render(<BetaBadge />);

    const span = container.querySelector('span');
    expect(span).toBeInTheDocument();
    expect(span?.textContent).toBe('Beta');
  });

  it('has correct styling classes', () => {
    const { container } = render(<BetaBadge />);

    const span = container.querySelector('span');
    expect(span).toHaveClass('items-center');
    expect(span).toHaveClass('px-2');
    expect(span).toHaveClass('py-0.5');
    expect(span).toHaveClass('rounded-full');
    expect(span).toHaveClass('text-xs');
    expect(span).toHaveClass('font-medium');
    expect(span).toHaveClass('bg-yellow-100');
    expect(span).toHaveClass('text-yellow-800');
  });

  it('matches snapshot', () => {
    const { container } = render(<BetaBadge />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
