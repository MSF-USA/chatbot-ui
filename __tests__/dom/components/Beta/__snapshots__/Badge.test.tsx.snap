// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BetaBadge > matches snapshot 1`] = `
<span
  class="items-center px-2 py-0.5 rounded-full text-xs font-medium bg-yellow-100 text-yellow-800"
>
  Beta
</span>
`;
